### PR TITLE
new style prod function

### PR DESCRIPTION
**Pull request by kmaehashi**
This PR implements new style `prod` function.

~~This PR depends on `F.repeat` proposed in PR #3735.~~ (merged)

### CODE REVIEW

**Comment by kmaehashi** [comment]
Remove unused `retry` decorator and confirmed 10000 repeats.

**Comment by takagi** [comment]
Would you resolve the conflicts?

**Comment by kmaehashi** [comment]
Resolved conflict.

**Comment by takagi** [comment]
LGTM, will merge after CI passes!

**Comment by takagi** [comment]
jenkins, test this please.

**Comment by kmaehashi** [comment]
`F.repeat` is now merged. Resolved conflict.

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
@kmaehashi Could you check @toslunar's comment?

File: chainer/functions/math/prod.py [hunk 1]
```diff
@@ -79,13 +82,18 @@ def backward(self, x, gy):
         x = x.transpose(transpose_axes)
         transposed_shape = x.shape
         x = x.reshape(-1, n_reduced_elements)
-        extended_x = xp.repeat(x, n_reduced_elements, 0)
-        mask = xp.tile(xp.arange(n_reduced_elements), n_output_elements)
-        extended_x[xp.arange(x.size), mask] = 1
-
-        gx = extended_x.prod(1)
+        extended_x = chainer.functions.repeat(x, n_reduced_elements, 0)
+        mask = chainer.functions.tile(
```
[the remark this answers is not in the document]
Comment (kmaehashi) [reply chainer/functions/math/prod.py]: Thanks, fixed!

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
LGTM!

**Comment by kmaehashi** [comment]
@niboshi I mistakenly removed the backport label. Could you make a backport PR?

**Comment by niboshi** [comment]
@kmaehashi Sorry I've made the backport.

**Comment by kmaehashi** [comment]
Thanks!

**Pull request merged**